### PR TITLE
spi: Add read/write register wrapper

### LINKED ISSUE #17: Wrap mfrc522 register read/write

### PR DESCRIPTION
**Pull request by Skallwar**
Fixes #17

### CODE REVIEW

File: module/mfrc522_spi.c [hunk 1]
```diff
@@ -11,3 +11,24 @@ struct address_byte address_byte_build(u8 mode, u8 addr)
...
+	return value;
+}
+
+void mfrc522_register_write(struct spi_device *client, u8 reg, u8 value)
+{
+	struct address_byte reg_write =
+		address_byte_build(MFRC522_SPI_WRITE, reg);
+
+	spi_write(client, &reg_write, 1);
+	spi_write(client, &value, 1);
```
Comment (CohenArthur) [inline module/mfrc522_spi.c:33]: Maybe we could create an array and only do one call to `spi_write`?
```suggestion
#define REG_WRITE_LEN 2
        u8 tx_buffer[REG_WRITE_LEN] = { reg_write, value };
	spi_write(client, tx_buffer, REG_WRITE_LEN);
#undef REG_WRITE_LEN
```
This way we only need to acquire the various SPI ressources such as locks once

File: module/mfrc522_spi.c [hunk 2]
```diff
@@ -11,3 +11,24 @@ struct address_byte address_byte_build(u8 mode, u8 addr)
...
 }
+
+u8 mfrc522_register_read(struct spi_device *client, u8 reg)
+{
+	struct address_byte reg_read =
+		address_byte_build(MFRC522_SPI_READ, reg);
+	u8 value;
+
+	spi_write(client, &reg_read, 1);
+	spi_read(client, &value, 1);
```
Comment (CohenArthur) [inline module/mfrc522_spi.c:22]: Like below, we could also do a `spi_write_then_read()`
```suggestion
        spi_write_then_read(client, &reg_read, 1, &value, 1);
```

File: module/mfrc522_spi.c (hunk 1 above)
Comment (Skallwar) [reply module/mfrc522_spi.c:33]: Good idea

File: module/mfrc522_spi.c [hunk 3]
```diff
@@ -18,17 +20,18 @@ u8 mfrc522_register_read(struct spi_device *client, u8 reg)
...
 }
 
 void mfrc522_register_write(struct spi_device *client, u8 reg, u8 value)
 {
 	struct address_byte reg_write =
 		address_byte_build(MFRC522_SPI_WRITE, reg);
+	u8 data[2] = { 0, value };
+
+	// We cannot directly put reg_write in data[0] at init time
+	memcpy(&data[0], &reg_write, sizeof(u8));
```
Comment (CohenArthur) [inline module/mfrc522_spi.c:34]:
```suggestion
	data[0] = reg_write;
```
?

**Review by CohenArthur — APPROVED** [review]

File: module/mfrc522_spi.c (hunk 3 above)
Comment (Skallwar) [reply module/mfrc522_spi.c:34]: Not working:
```
mfrc522_spi.c:30:2: error: aggregate value used where an integer was expected
```

Comment (CohenArthur) [reply module/mfrc522_spi.c:34]: Oh my bad I had completely forgotten that we had a struct address byte. I thought we were still dealing with u8. Perfect then! Good job and thanks

File: module/mfrc522_spi.c [hunk 4]
```diff
@@ -11,3 +13,25 @@ struct address_byte address_byte_build(u8 mode, u8 addr)
 
 	return byte;
 }
+
+u8 mfrc522_register_read(struct spi_device *client, u8 reg)
+{
+	struct address_byte reg_read =
```
Comment (n1tram1) [inline module/mfrc522_spi.c:19]: this is not very gnu89 of you

File: module/mfrc522_spi.c [hunk 5]
```diff
@@ -11,3 +13,25 @@ struct address_byte address_byte_build(u8 mode, u8 addr)
...
+		address_byte_build(MFRC522_SPI_READ, reg);
+	u8 value;
+
+	spi_write_then_read(client, &reg_read, 1, &value, 1);
+	return value;
+}
+
+void mfrc522_register_write(struct spi_device *client, u8 reg, u8 value)
+{
+	struct address_byte reg_write =
```
Comment (n1tram1) [inline module/mfrc522_spi.c:29]: not gnu89

**Review by n1tram1 — CHANGES_REQUESTED** [review]

**Review by n1tram1 — APPROVED** [review]

**Pull request merged**